tests/BasicSearchTests.FunctionalTests.Core/EnvironmentSettings.cs

```diff
@@ -61,6 +62,19 @@ public static string ConfigurationName
             }
         }
 
+        public static string Slot
+        {
+            get
+            {
+                if (string.IsNullOrEmpty(_slot))
```

Comment:
This could have been an instance of `Lazy<string>`, then you'd need not to write the code to check if it was initialized.